csharp/src/Ice/TraceUtil.cs

```diff
@@ -2,315 +2,190 @@
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
 
-using ZeroC.Ice;
+using System;
 using System.Collections.Generic;
-using System.Diagnostics;
 using System.Globalization;
+using System.IO;
 
-namespace IceInternal
+namespace ZeroC.Ice
 {
     internal static class TraceUtil
     {
-        internal static void TraceSend(Communicator communicator, IList<System.ArraySegment<byte>> buffer)
+        internal static void TraceFrame(Communicator communicator, Span<byte> header, OutgoingRequestFrame frame) =>
```

Comment:
`Span<byte>` should be `ReadOnlySpan<byte>`, but more importantly, is this header parameter temporary until we refactor more this code?

The frame doesn't hold the information we extract out of header because it's not or shouldn't be relevant to the frame.

Reply:
We have to decided, how we want to pass the size, requetId and compressionStatus

The size can be calculated from the frame (frame.Size + HeaderSize + 4), so maybe we can remove that.

For compressionStatus/requestId we can pass them explicitly or read them from the header, but they don't belong to the frame class

Reply:
Fixed methods to use `ReadOnlySpan<byte>`